arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1674,9 +1690,9 @@ int ClusterInfo::createCollectionCoordinator(std::string const& databaseName,
           if (!tmpError.empty()) {
             *errMsg = "Error in creation of collection:" + tmpError + " "
                 + __FILE__ + std::to_string(__LINE__);
-            *dbServerResult = TRI_ERROR_CLUSTER_COULD_NOT_CREATE_COLLECTION;
+            dbServerResult->store(TRI_ERROR_CLUSTER_COULD_NOT_CREATE_COLLECTION, std::memory_order_release);
           } else {
-            *dbServerResult = setErrormsg(TRI_ERROR_NO_ERROR, *errMsg);
+            dbServerResult->store(setErrormsg(TRI_ERROR_NO_ERROR, *errMsg), std::memory_order_release);
```

Comment:
Doesn't this still allow unprotected access to `errorMsg`?

Reply:
I don't think so. We first set `errMsg` before setting `dbServerResult` with `memory_order_release`. Later we load `dbServerResult` with `memory_order_acquire`, and only if it has a value >= 0 (i.e., we read a value stored here), then we access `errMsg`. So the acquire/release semantic for the operations on `dbServerResult` orders the operations on `errMsg`.

Reply:
_errMsg_ is written before dbServerResult is set, so there should be a happens-before relationship when the other thread tries to read _errMsg_